service/src/allocation_monitor.rs

```diff
@@ -24,44 +21,42 @@ struct AllocationMonitorInner {
     interval_ms: u64,
     graph_network_id: u64,
     eligible_allocations: Arc<RwLock<Vec<Allocation>>>,
-    attestation_signers: Arc<RwLock<HashMap<Address, AttestationSigner>>>,
-    indexer_mnemonic: String,
-    chain_id: U256,
-    dispute_manager: Address,
+    watch_sender: Sender<()>,
+    watch_receiver: Receiver<()>,
 }
 
+#[cfg_attr(test, faux::create)]
 #[derive(Debug, Clone)]
 pub struct AllocationMonitor {
-    monitor_handle: Arc<tokio::task::JoinHandle<()>>,
+    _monitor_handle: Arc<tokio::task::JoinHandle<()>>,
```

Comment:
what do we expect this to do? 

Is it right to think of `AllocationMonitorInner` as something that just keeps information including watcher and receiver while `AllocationMonitor` is doing the actual monitoring?

Reply:
The main reason I created an "inner" is to make this possible:
https://github.com/graphops/indexer-service-rs/blob/a410943128cabe8166880b041bb287714e770db0/service/src/allocation_monitor.rs#L58-L63

Basically I needed the spawned thread not depend on `AllocationMonitor`, because it is itself belonging to `AllocationMonitor`.
I could just clone all the values that the thread needs, but the "inner" makes it visually cleaner. Also makes it cleaner to Arc clone `AllocationMonitor`.

tl;dr it's mostly easthetics, I guess

Reply:
As for the handle, I don't like the idea of not explicitely keeping it somewhere. Is the thread even guaranteed to continue existing if the handle is lost? (I actually don't know yet what's going on at a lower level here)

Reply:
And you are keeping track of the handle so it won't "roam free"? Would we need to specifically shutdown the spawned threads when we get termination or exit signal?

Reply:
> Would we need to specifically shutdown the spawned threads when we get termination or exit signal?

Well, that's a good question. I guess we'll see when we tie everything together :grin:

Reply:
Calling the Rust masters! Would @axiomatic-aardvark know?

Reply:
I think it makes sense either way, the thread will continue even without the handle, but it does seem more idiomatic to keep track of it. And as @hopeyen suggested that allows for a shutdown signal (maybe a `Arc<AtomicBool>`) to be used to break out of the loop